app/src/main/res/layout/view_site_protections_toggle.xml

```diff
@@ -0,0 +1,87 @@
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:background="@drawable/background_site_protections_toggle">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <com.duckduckgo.mobile.android.ui.view.text.DaxTextView
+            android:id="@+id/protectionsSwitchLabel"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/keyline_4"
+            android:layout_marginTop="@dimen/keyline_4"
+            android:text="@string/brokenSiteProtectionsOff"
+            app:layout_constraintEnd_toStartOf="@+id/protectionsSwitch"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:text="@string/brokenSiteProtectionsOn" />
+
+        <com.duckduckgo.mobile.android.ui.view.SwitchView
+            android:id="@+id/protectionsSwitch"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="@dimen/keyline_4"
+            app:layout_constraintBottom_toBottomOf="@id/protectionsSwitchLabel"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@+id/protectionsSwitchLabel"
+            app:layout_constraintTop_toTopOf="@id/protectionsSwitchLabel"
+            tools:checked="true" />
+
+        <FrameLayout
+            android:id="@+id/protectionsBannerMessageContainer"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/keyline_3"
+            android:paddingStart="@dimen/keyline_4"
+            android:paddingTop="@dimen/keyline_3"
+            android:paddingEnd="@dimen/keyline_4"
+            android:paddingBottom="@dimen/keyline_4"
+            android:visibility="gone"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/protectionsSwitchLabel"
+            tools:visibility="visible">
+
+            <com.duckduckgo.mobile.android.ui.view.text.DaxTextView
+                android:id="@+id/protectionsBannerMessage"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                tools:text="@string/brokenSiteProtectionsOnBannerMessage" />
+
+        </FrameLayout>
+
+        <!-- This is just to keep the correct bottom padding when the banner is not visible -->
```

Comment:
Where is this needed?
Also, do you need the outer FrameLayout?

Reply:
It makes the top and bottom margins symmetrical when `protectionsBannerMessageContainer` has visibility set to `GONE`. I agree that purpose of this view is very non-obvious (which is why I added this comment), so I had a second look at this and refactored it here: https://github.com/duckduckgo/Android/pull/3774/commits/21db22d84b8c114a87afede870985f7593a18fcf

The top level FrameLayout is not needed, removed here: https://github.com/duckduckgo/Android/pull/3774/commits/5f48e0cb3d2d9ca31196e9d62686d4ea97f75a95